src/datasets/domain/models/Dataset.ts

```diff
@@ -0,0 +1,40 @@
+export interface Dataset {
+  id: number;
+  persistentId: string;
+  versionId: number;
+  versionInfo: DatasetVersionInfo;
+  license: DatasetLicense;
+  metadataBlocks: DatasetMetadataBlock[];
+}
+
+export interface DatasetVersionInfo {
+  majorNumber: number;
+  minorNumber: number;
+  state: DatasetVersionState;
+  createTime: Date;
+  lastUpdateTime: Date;
+  releaseTime?: Date;
+}
+
+export enum DatasetVersionState {
+  DRAFT = 'DRAFT',
+  RELEASED = 'RELEASED',
```

Comment:
No - only one state. The code looks redundant to me. All of those calls look at the workingVersion.versionState which is an enum - so if DatasetPage.workingVersion.released is true, it can't be draft or deaccessioned. inReview is slightly different - it's not one of the enum values, but you can only be inReview if the dataset is draft.